src/MemoryGraph/MemoryGraph.cs

```diff
@@ -146,7 +161,7 @@ void IFastSerializable.FromStream(Deserializer deserializer)
 
         // This array survives after the constructor completes
         // TODO Fold this into the existing blob. Currently this dominates the Size cost of the graph!
-        protected GrowableArray<Address> m_nodeAddresses;
+        protected IMutableList<Address> m_nodeAddresses;
```

Comment:
💡 I believe performance would be improved by simply using `SegmentedList<Address>` here and removing the ability to use `GrowableList<Address>`

Reply:
I'm not convinced the indirection is observably more costly than the used of interface dispatch. 😄